firebase-firestore/src/main/java/com/google/firebase/firestore/local/IndexManager.java

```diff
@@ -81,9 +80,15 @@
   /** Returns the documents that match the given target based on the provided index. */
   Set<DocumentKey> getDocumentsMatchingTarget(FieldIndex fieldIndex, Target target);
 
-  /** Returns the next collection group to update. */
+  /**
+   * Returns the next collection group to update. Returns null after reaching the provided max
+   * sequence number.
+   */
   @Nullable
-  String getNextCollectionGroupToUpdate(Timestamp lastUpdateTime);
+  String getNextCollectionGroupToUpdate(int currentMaxSequenceNumber);
+
+  /** Returns the highest sequence number in the collection groups table. */
+  int getMaxCollectionGroupSequenceNumber();
```

Comment:
getIndexBackfillSequenceNumber? I think it might be more interesting to say what the sequence number is for.

I am 50/50 on whether we need this method in our interface though.